### PR TITLE
Backport 4.x deprecations for easier upgrading.

### PR DESCRIPTION
**Pull request by dereuromark**
Maybe someone can  help with it, or add more missing ones?
So far I extracted the obvious ones from 4.x migration guide here.

### CODE REVIEW

File: src/View/Helper/UrlHelper.php [hunk 1]
```diff
@@ -48,6 +48,9 @@ public function build($url = null, $options = false)
             'escape' => true,
         ];
         if (!is_array($options)) {
+            deprecationWarning(
+                'The `bool` array option is deprecated. Use an array here instead with `\'fullBase\' => true`.'
+            );
```
Comment (markstory) [inline src/View/Helper/UrlHelper.php]: This one causes a ton of failures in the test suite, and I'm not sure it is a high value change to be making this late in 3.x's life. It also doesn't add a significant burden to us to maintain going forwards.

Comment (dereuromark) [reply src/View/Helper/UrlHelper.php]: You mean we shouldnt throw the deprecation warning?

Comment (dereuromark) [reply src/View/Helper/UrlHelper.php]: I added it to shim plugin instead: https://github.com/dereuromark/cakephp-shim/pull/53

Comment (markstory) [reply src/View/Helper/UrlHelper.php]: Yes, if this deprecation causes many of our tests to fail it will likely be tedious for application developers too.

Comment (saeideng) [reply src/View/Helper/UrlHelper.php]: I'm :+1:  with dont adding more deprecation warning to cake 3.8
we can add it since cake 4.0

**Comment by dereuromark** [comment]
Is this ok?

**Pull request merged**